### DISCUSSION TITLE
vipsthumbnail - resize PNG without antialias

### DISCUSSION THREAD

**Discussion by mistermiracl** [libvips/libvips#3083] [Q&A]
I have an image, a triangle in this case, that is using just one color, meaning it doesn't have any pixels with alpha pixels except for the background that is fully transparent, I would like to keep it that way

og image
![triangle](https://user-images.githubusercontent.com/97761926/194470001-b047fcc8-c39d-43b2-a526-f997bddb3430.png)

 but when I resize the image using the following command:

`vipsthumbnail triangle.png -s 500x -o triangle-tailored.png`

antialias pixels are added to sharpen the edges Im guessing
![triangle-resized](https://user-images.githubusercontent.com/97761926/194470096-ddb570a4-3102-4eea-9425-f264f0b2f425.png)

is it possible to not do that using `vipsthumbnail`? thanks

**Comment by jcupitt** [comment]
Hi @mistermiracl,

No, `vipthumbnail` doesn't let you select the interpolator -- it tries to do this for you automatically..

You'll need to use one the language bindings (python, ruby, C++, php, rust, etc.) if you want more control over resizing.

**Reply by mistermiracl** [reply]
Hey @jcupitt, I can do that, in this case thou, which interpolator would do in order to disable antialiasing?

**Reply by jcupitt** [reply]
It depends on your needs. Are these images the output of a classifier, for example? Then you'd probably want `nearest`. If you are just interested in avoiding edge interpolation, then a block filter followed by nearest would be more accurate.

**Reply by sergeevabc** [reply]
@jcupitt, is there any workaround for ordinary users, e.g. using `vips` instead of `vipsthumbnail`?

**Reply by jcupitt** [reply]
You can use `resize` instead, eg.:

```
$ vips resize triangle.png x.png 0.5 --kernel nearest
```

To make:

![x](https://github.com/libvips/libvips/assets/580843/021a23c3-40d3-4b43-8184-b72922d7a91e)


The `0.5` is the resize ratio, so shrink by x2 in this case.

**Reply by sergeevabc** [reply]
> @jcupitt
> If you are just interested in avoiding edge interpolation, then a block filter followed by nearest would be more accurate.

Which one is a block filter here?

``` console
$ vips resize
...
   kernel       - Resampling kernel, input VipsKernel
                        default enum: lanczos3
                        allowed enums: nearest, linear, cubic, mitchell, lanczos2, lanczos3
...
```

**Reply by sergeevabc** [reply]
Also, working with the `resize` command has the disadvantage that I rarely know the scale factor. But I  know what width or height is desired. For example, the following is possible with ImageMagick: `magick in.png -interpolate integer -filter point -resize 600x out.png`

**Reply by jcupitt** [reply]
The block filter is a separate operation -- `shrink` -- due to the very different implementation.

You can compute scale factors for `resize` in bash with eg.:

```shell
width=$(vipsheader -f width $infile)
scale=$(echo $(echo "scale=10; $width / $target_size" | bc))
```

Though it's a bit ugly. Something like python is a lot simpler (obviously):

```python
scale = image.width / target_size
```
